Frontend последняя версия

tanyginvv opened the pull request.

tanyginvv committed:
добавлена логика лайков и избранных+ рецепт дня

tanyginvv committed:
Добавил последнюю версию фронта  с последней версией бэка

tanyginvv commented:
Запуск фронтенда с папки /Recipes_frontend
сначала npm i для скачивания зависимостей node_modules,
затем npm run dev для запуска приложения

tanyginvv committed:
добавил тест кейс для создания рецепта

VladDotMe commented:
Когда открываешь главную страницу приложения, показывается ошибка, хотя должна открываться homePage. 
Я бы вообще убрал /homePage и оставил, что по умолчанию всегда открывается эта страница.

![image](https://github.com/user-attachments/assets/0a6bf99a-5a1d-4d0e-beb4-066988fb485b)

VladDotMe commented:
Нужно сделать, чтобы frontend отдавался с backend'a.
Тогда не нужно будет отдельно запускать frontend. 
Посмотри пример моего проекта с ангуляром, который я тебе в лс скидывал, там такое есть, ну или погугли как это делается с реактом)

VladDotMe commented:
Пока больше комментарии по UI/UX., код не сильно смотрел (т.к. в реакте мало разбираюсь), попозже код гляну..

VladDotMe commented:
- Теги возвращаются рандомные, а хочется чтобы были самые популярные (например).
- Два запроса /recipes, не очень круто.. Лучше обернуть во враппер(еще одну DTO) список рецептов и добавить туда булеан типа "isNextPageAvailable"

VladDotMe commented:
Еще если удалить какой-то рецепт, то может задублироваться список рецептов или еще какая-нибудь непредсказуемая штука случится) Есть какие-нибудь идеи как можно предотвратить?

VladDotMe commented:
Запрос на /users/ некорректный, т.к. название endpoint'a в множественном числе и значит /users/ должен был бы возвращать список юзеров, а не текущего. 

Лучше изменить название метода на бекенде на /current-user или что-то типа того

VladDotMe commented:
Аналогично с PUT методом, лучше назвать /users/current-user

VladDotMe commented:
![image](https://github.com/user-attachments/assets/6af5104a-1b38-4c7b-9521-2d412380060f)
Тут сразу две проблемы:
1. hash в названии, хотя это не хеш (да и решили хеши не использовать для передачи, т.к. https достаточно защищен)
2. на UI нет валидации того, что пароли одинаковые.

VladDotMe commented:
![image](https://github.com/user-attachments/assets/f756dc31-68b5-470c-b4cf-df1dad17108b)

hash в названии, хотя это не хеш (да и решили хеши не использовать для передачи, т.к. https достаточно защищен)

VladDotMe commented:
Убрать на бекенде валидацию на непустой Description (он по умолчанию пустой, при редактировании значит тоже должна быть возможность отправить пустой)

VladDotMe commented:
В целом если меняю данные и ввожу что-то неверное, бекенд возвращает ошибку но на фронтенде написано, что все успешно.

![image](https://github.com/user-attachments/assets/9354388b-8cc5-4f5a-85cc-df3ee1d6ad90)

tanyginvv committed:
правки по комментам

1)Добавил enum для получения рецептов
2)добавил составные индексы для поиска лайков и избранных

tanyginvv committed:
букву не ту поставил

tanyginvv committed:
Merge branch 'backend-likes-and-favourites' into frontend-all

tanyginvv committed:
микро-правка

tanyginvv committed:
Правки по комментам

1) сделал возвращение на главную старницу при неправильном адресе
2)вывожу теперь популярные теги
3)поправил пагинацию, теперь один запрос
4)поправил end-pointы у апи пользователя
5)поправил логику с обновлением пользователя( пароли и описание)+ ошибки выводит если чтото не так при обновлении

tanyginvv committed:
обновления с ветки фронтенда

tanyginvv committed:
правки по комментам

tanyginvv committed:
Merge branch 'backend-likes-and-favourites' into frontend-all

tanyginvv committed:
вынес recipeList в компонент для переиспользования, обновил логику обновления пользователя + правки по бэку

tanyginvv committed:
Squashed commit of the following:

commit 052450b585fcb97e2edcded259d008828a04161b
Author: tanyginvv <tanyginvv@mail.ru>
Date:   Sun Sep 8 16:50:52 2024 +0300

    правки по атрибутам лайков и избранных

commit d76aa1ac51d9c00b0c3e0f6dc093162efdaea5e8
Author: tanyginvv <tanyginvv@mail.ru>
Date:   Sun Sep 8 15:05:45 2024 +0300

    правки по комментам

commit 8019e2a2f0b192d33d71c1a63ef20e7acd62eab7
Author: tanyginvv <tanyginvv@mail.ru>
Date:   Thu Sep 5 22:13:27 2024 +0300

    обновления с ветки фронтенда

commit 4d2b966429d1c97bea5fe8e57143842f489abe1f
Author: tanyginvv <tanyginvv@mail.ru>
Date:   Thu Sep 5 17:48:45 2024 +0300

    микро-правка

commit c688361a4e49a17ebb61306f9a652aec949a085a
Author: tanyginvv <tanyginvv@mail.ru>
Date:   Thu Sep 5 17:45:19 2024 +0300

    букву не ту поставил

commit 4dd571b71c408b16d73e1a4ce47c2d7fc5804d05
Author: tanyginvv <tanyginvv@mail.ru>
Date:   Thu Sep 5 17:43:22 2024 +0300

    правки по комментам

    1)Добавил enum для получения рецептов
    2)добавил составные индексы для поиска лайков и избранных

tanyginvv committed:
тесты

tanyginvv committed:
Merge branch 'master' into frontend-all

tanyginvv committed:
добавил проверку на кол-во шагов и ингредиентов

tanyginvv committed:
обновил тесты для кол-ва ингредиентов и шагов

tanyginvv committed:
добавлено логирование

tanyginvv committed:
Merge branch 'backend-tests' into frontend-all

tanyginvv committed:
сделал запуск фронта с бэком

tanyginvv committed:
поправлен фронт, страницы вынесены в отдельную папку
